hikari/presences.py

```diff
@@ -135,6 +148,117 @@ class ActivityAssets:
     small_text: typing.Optional[str] = attr.field(repr=True)
     """The text that'll appear when hovering over the small image, if set."""
 
+    def _make_asset_url(self, asset: typing.Optional[str], ext: str, size: int) -> typing.Optional[files.URL]:
+        if asset is None:
+            return None
+
+        try:
+            resource, identifier = asset.split(":", 1)
+            return files.URL(url=_DYNAMIC_URLS[resource].format(identifier))
+
+        except KeyError:
+            raise RuntimeError("Unknown asset type") from None
+
+        except ValueError:
+            assert self._application_id is not None
+            return routes.CDN_APPLICATION_ASSET.compile_to_file(
+                urls.CDN_URL,
+                application_id=self._application_id,
+                asset=asset,
+                size=size,
+                file_format=ext,
+            )
+
+    @property
+    def large_image_url(self) -> typing.Optional[files.URL]:
+        """Large image asset URL, if there is one.
+
+        Returns
+        -------
+        typing.Optional[hikari.files.URL]
+            The URL, or `builtins.None` if no large image asset exists.
+
+        Raises
+        ------
+        builtins.RuntimeError
```

Comment:
Maybe avoid putting a property for this one if it raises or just return None instead of raising